sys/net/routing/ng_rpl/ng_rpl_control_messages.c

```diff
@@ -0,0 +1,579 @@
+/*
+ * Copyright (C) 2013 - 2014  INRIA.
+ * Copyright (C) 2015 Cenk Gündoğan <cnkgndgn@gmail.com>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ *
+ * @author  Cenk Gündoğan <cnkgndgn@gmail.com>
+ */
+#include "net/ng_rpl.h"
+#include "inet_ntop.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+#if ENABLE_DEBUG
+static char addr_str[NG_IPV6_ADDR_MAX_STR_LEN];
+#endif
+
+#define NG_RPL_GROUNDED_SHIFT           (7)
+#define NG_RPL_MOP_SHIFT                (3)
+#define NG_RPL_OPT_DODAG_CONF_LEN       (14)
+#define NG_RPL_SHIFTED_MOP_MASK         (0x7)
+#define NG_RPL_PRF_MASK                 (0x7)
+
+void _ng_rpl_send(ng_pktsnip_t *pkt, ng_ipv6_addr_t *src, ng_ipv6_addr_t *dst)
+{
+    ng_pktsnip_t *hdr;
+    ng_ipv6_addr_t all_RPL_nodes = NG_IPV6_ADDR_ALL_RPL_NODES;
+    kernel_pid_t iface = ng_ipv6_netif_find_by_addr(NULL, &all_RPL_nodes);
+    if (iface == KERNEL_PID_UNDEF) {
+        DEBUG("RPL: no suitable interface found for this destination address\n");
+        ng_pktbuf_release(pkt);
+        return;
+    }
+
+    if (src == NULL) {
+        ng_ipv6_addr_t ll_addr;
+        ng_ipv6_addr_set_link_local_prefix(&ll_addr);
+        src = ng_ipv6_netif_match_prefix(iface, &ll_addr);
+        if (src == NULL) {
+            DEBUG("RPL: no suitable src address found for this destination address\n");
+            ng_pktbuf_release(pkt);
+            return;
+        }
+    }
+
+    if (dst == NULL) {
+        dst = &all_RPL_nodes;
+    }
+
+    hdr = ng_ipv6_hdr_build(pkt, (uint8_t *)src, (src ? sizeof(ng_ipv6_addr_t) : 0),
```

Comment:
good point! @authmillenon also pointed this out in an offline discussion, but I completely forgot about this. thanks for the reminder. Will update.